src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -723,9 +724,18 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
 
       // Create the cluster
       createClusterConfig = CreateClusterConfig(machineConfig, initScript, serviceAccountInfo.clusterServiceAccount, credentialsFileName, stagingBucket, clusterScopes, clusterVPCSettings, clusterRequest.properties)
-      operation <- gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      retryResult <- retryExponentially(whenGoogleZoneCapacityIssue, "Cluster creation failed because zone with adequate resources was not found") { () =>
+        gdDAO.createCluster(googleProject, clusterName, createClusterConfig)
+      }
+      operation <- retryResult match {
+        case Right(value) => Future.successful(value._2)
```

Comment:
readability nit: you can do `case Right((_, op)) => Future.successful(op)` here to name the value in the tuple that you're using